src/__tests__/extensions/replay/sessionrecording.test.ts

```diff
@@ -896,21 +898,41 @@ describe('SessionRecording', () => {
                 expect(assignableWindow.rrweb.record.takeFullSnapshot).not.toHaveBeenCalled()
             })
 
-            it('does not send a full snapshot if there is not a new session or window id', () => {
+            it('does not send a full snapshot if there is not a new session or window id and the session has a snapshot', () => {
                 assignableWindow.rrweb.record.takeFullSnapshot.mockClear()
 
-                // we always take a full snapshot when there hasn't been one
-                // and use _fullSnapshotTimer to track that
-                // we want to avoid that behavior here, so we set it to any value
-                sessionRecording['_fullSnapshotTimer'] = 1
-
                 sessionIdGeneratorMock.mockImplementation(() => 'old-session-id')
                 windowIdGeneratorMock.mockImplementation(() => 'old-window-id')
                 sessionManager.resetSessionId()
+                sessionRecording['lastSessionFullSnapshot'] = 'old-session-id'
 
                 _emit(createIncrementalSnapshot())
                 expect(assignableWindow.rrweb.record.takeFullSnapshot).not.toHaveBeenCalled()
             })
+
+            it('does not send a full snapshot if the session has no full snapshot', () => {
```

Comment:
There is no `not` in this expect. Should the test be renamed?
```suggestion
            it('does send a full snapshot if the session has no full snapshot', () => {
```